vcu/Src/drive_by_wire_mock.c

```diff
@@ -442,67 +443,62 @@ static const CLI_Command_Definition_t mcInitCommandDefinition =
 
 
 BaseType_t setInverterParameter(char *writeBuffer, size_t writeBufferLength, const char *commandString){
-    uint16_t parameterAddress;
-    uint16_t newValue;
+    uint16_t parameterAddress=0;
+    uint16_t newValue=0;
     BaseType_t status;
     uint16_t readValue = 0;
     uint32_t currentState = fsmGetState(&fsmHandle); //returns an index
-    BaseType_t paramLen;
+    BaseType_t paramLen = 0;
 
     const char *addressParam = FreeRTOS_CLIGetParameter(commandString, 1, &paramLen);
     const char *valueParam = FreeRTOS_CLIGetParameter(commandString, 2, &paramLen);
     
     sscanf(addressParam, "%hu", &parameterAddress);
     sscanf(valueParam, "%hu", &newValue);
 
-    if (HV_Power_State == HV_Power_State_On){ 
+    if (getHvEnableState()){ 
         COMMAND_OUTPUT("Error: Car is at high voltage. Operation not permitted\n");
         return pdFALSE;
     }
     else if (currentState == STATE_EM_Enable){ 
         COMMAND_OUTPUT("Error: Car is in drive mode. Operation not permitted\n");
         return pdFALSE;
-    } else{
+    }
+    if (turnOnMotorController() != HAL_OK) {
+        COMMAND_OUTPUT("MC failed to turn on\n");
+        return pdFALSE;
+    }
 
+    status = mcWriteParamCommand(parameterAddress, newValue); 
 
-        status = mcWriteParamCommand(parameterAddress, newValue); 
-        if (status == HAL_OK){
-            COMMAND_OUTPUT("Parameter written successfully. \n");
-        }
-        if (status != HAL_OK){
-            COMMAND_OUTPUT("Failed to write parameter message at address %u\n", parameterAddress);
-            return pdFALSE;
-        }
+    if (status != HAL_OK){
+        COMMAND_OUTPUT("Failed to write parameter message at address %u\n", parameterAddress);
+        return pdFALSE;
+    }
 
-        HAL_Delay(1000);
+    turnOffMotorControllers();
+    turnOnMotorController();
 
-        status = mcReadParamCommand(parameterAddress, readValue);
-        if (status == HAL_OK){
-            COMMAND_OUTPUT("Parameter read successfully. \n");
+    status = mcReadParamCommand(parameterAddress, readValue);
```

Comment:
readValue here does not actually change it remains whatever you set it to (look at the function). take a look at the comment I left in Dorris' PR